nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/dynamodb/GetDynamoDB.java

```diff
@@ -100,42 +104,75 @@
     }
 
     @Override
-    public void onTrigger(final ProcessContext context, final ProcessSession session) {
-        List<FlowFile> flowFiles = session.get(context.getProperty(BATCH_SIZE).evaluateAttributeExpressions().asInteger());
-        if (flowFiles == null || flowFiles.size() == 0) {
-            return;
-        }
+    public List<ConfigVerificationResult> verify(final ProcessContext context, final ComponentLog verificationLogger, final Map<String, String> attributes) {
+        final List<ConfigVerificationResult> results = new ArrayList<>(super.verify(context, verificationLogger, attributes));
 
-        Map<ItemKeys,FlowFile> keysToFlowFileMap = new HashMap<>();
+        final TableKeysAndAttributes tableKeysAndAttributes = getTableKeysAndAttributes(context, attributes);
 
         final String table = context.getProperty(TABLE).evaluateAttributeExpressions().getValue();
-        TableKeysAndAttributes tableKeysAndAttributes = new TableKeysAndAttributes(table);
-
-        final String hashKeyName = context.getProperty(HASH_KEY_NAME).evaluateAttributeExpressions().getValue();
-        final String rangeKeyName = context.getProperty(RANGE_KEY_NAME).evaluateAttributeExpressions().getValue();
         final String jsonDocument = context.getProperty(JSON_DOCUMENT).evaluateAttributeExpressions().getValue();
 
-        for (FlowFile flowFile : flowFiles) {
-            final Object hashKeyValue = getValue(context, HASH_KEY_VALUE_TYPE, HASH_KEY_VALUE, flowFile);
-            final Object rangeKeyValue = getValue(context, RANGE_KEY_VALUE_TYPE, RANGE_KEY_VALUE, flowFile);
+        if (tableKeysAndAttributes.getPrimaryKeys().isEmpty()) {
 
-            if ( ! isHashKeyValueConsistent(hashKeyName, hashKeyValue, session, flowFile)) {
-                continue;
-            }
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(Outcome.SKIPPED)
+                    .verificationStepName("Get DynamoDB Items")
+                    .explanation(String.format("Skipped getting DynamoDB items because no primary keys would be included in retrieval"))
+                    .build());
+        } else {
+            try {
+                final DynamoDB dynamoDB = getDynamoDB(getConfiguration(context).getClient());
+                int totalCount = 0;
+                int jsonDocumentCount = 0;
 
-            if ( ! isRangeKeyValueConsistent(rangeKeyName, rangeKeyValue, session, flowFile) ) {
-                continue;
-            }
+                BatchGetItemOutcome result = dynamoDB.batchGetItem(tableKeysAndAttributes);
```

Comment:
Would it be better to do some sort of "DynamoDB Table Exists" rather than fetching data - this could potentially be costly if a lot of data is pulled back over the internet?

Reply:
Same comment as above with FetchS3Object -- unfortunately the DynamoDB client doesn't provide such a check (the `getTable` method doesn't actually make a call, only returns a new `Table` object).  I'd be inclined to leave it as is.

Reply:
Shame that `GetTable` does that, but oh well

Could we instead limit the pull of data to a single item then maybe, i.e. if we have to try fetching something, then fetch as little as possible in order to reduce both cost and time taken for the verification steps

Reply:
I'll see if I can decompose it more, but I think a single batch request is as small as we can do while still providing meaningful feedback during verification.

Reply:
Ok, I did some testing and I think this is still a reasonable implementation of the verification.  When presented with the verification screen, the user is prompted to fill in a single hash key value and a single range key value (assuming they keep the default EL expressions for those fields).  Since these will look up an item by primary key, the result of verification should be at most one item returned.

However, the extra testing here helped me find a NPE and improve the verification messaging in cases where the key values are inconsistent with the table key configuration.  I've pushed the changes with these improvements.

Reply:
I understand Verification better now thanks to a [Slack discussion](https://apachenifi.slack.com/archives/C0L9S92JY/p1638744006088000), so this is less of a concern as you're right that the user has the control over when this happens and against specific bits of data... so this check seems fine now, thanks for explaining!